### PR TITLE
:sparkles: feat:  support to set client in proxy mw

### PR DESCRIPTION
**Pull request by li-jin-gou**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. 
Explain the *details* for making this change. What existing problem does the pull request solve?

Fixes https://github.com/gofiber/fiber/issues/2116
- In some scenarios the user need set up some complex client config via custom client
- The case of a global client only gives the user no way to set up special configurations for only one middleware

so this pr allow to use custom client and  distinguish global client and custom client in proxy mw and compatible with previous logic.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x]  update of the docs repository

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by ReneWerner87** [comment]
please don´t forget 

- [ ] the update of the docs repository

**Comment by li-jin-gou** [comment]
> please don´t forget
> 
> * [ ]  the update of the docs repository

Thanks for reminding. :sob:

File: middleware/proxy/proxy.go [hunk 1]
```diff
@@ -24,34 +24,37 @@ func Balancer(config Config) fiber.Handler {
...
-	for _, server := range cfg.Servers {
-		if !strings.HasPrefix(server, "http") {
-			server = "http://" + server
-		}
+	var lbc = &fasthttp.LBClient{}
+	if config.Client == nil {
+		// Set timeout
+		lbc.Timeout = cfg.Timeout
+		// Scheme must be provided, falls back to http
+		for _, server := range cfg.Servers {
```
Comment (ReneWerner87) [inline middleware/proxy/proxy.go:32]: we should make clear in the readme or readme configuration that if the clients are set, then the server addresses and read and write buffer settings are not processed

Comment (li-jin-gou) [reply middleware/proxy/proxy.go:34]: done

**Pull request merged**